### PR TITLE
Liftwing conversion

### PR DESCRIPTION
**Pull request by justiceotuya**
NOTE: Please review the pull request process before opening your first PR: https://github.com/WikiEducationFoundation/WikiEduDashboard/blob/master/CONTRIBUTING.md#pull-request-process

## What this PR does
This MR fixes this issue https://github.com/WikiEducationFoundation/WikiEduDashboard/issues/5492 where Liftwing Api is converted from the previous Ores implementation

## Screenshots
Before:

Screenshot 2023-10-07 at 4 53 56 PM

After:

Screenshot 2023-10-07 at 4 50 37 PM

## Open questions and concerns
- learnt how to read and understand a ruby class
- learnt how to read data in a ruby class from the front end and how to also send data back to the frontend

### CODE REVIEW

**Comment by ragesoss** [comment]
I'd strongly prefer that the article quality data be fetched directly from the client, rather than hitting a Rails controller and having the server make the LiftWing requests.

If you ran into trouble with making the requests via JavaScript, post the details and maybe we can figure out how to fix it.

**Comment by justiceotuya** [comment]
ok. on it

**Comment by justiceotuya** [comment]
@ragesoss i am having issues sending data to the api.wikimedia endpoint to get the lifting data
This is my curl request. it works on the terminal but gets a COR error while using the browser and that was why I resorted in going via the server route.

one solution to this is to 
- use a proxy solution like ngrok (this means when testing ngrok must be used)
- whitelist localhost to be able to hit and get data from the api.wikimedia service

```bash
curl --location 'https://api.wikimedia.org/service/lw/inference/v1/models/enwiki-articlequality:predict' \
--header 'Content-Type: application/json' \
--data '{
    "rev_id": 1162049025
}'
```


attached is a screenshot of the error

SCR-20231009-roqy

**Comment by ragesoss** [comment]
Can you post the code for the request that resulted in that error?

**Comment by justiceotuya** [comment]
> Can you post the code for the request that resulted in that error?

Here  @ragesoss  https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5504

**Comment by ragesoss** [comment]
Try adding `&origin=*` to the end of the URL: https://www.mediawiki.org/wiki/API:Cross-site_requests#CORS_usage

**Comment by justiceotuya** [comment]
> Try adding `&origin=*` to the end of the URL: https://www.mediawiki.org/wiki/API:Cross-site_requests#CORS_usage

@ragesoss  same issue. i used

`https://api.wikimedia.org/service/lw/inference/v1/models/enwiki-articlequality:predict&origin=*`

also

`https://api.wikimedia.org/service/lw/inference/v1/models/enwiki-articlequality:predict&origin=*`

The error still persists

image

image

**Comment by ragesoss** [comment]
The Wikimedia ML team is looking into making this work: https://phabricator.wikimedia.org/T348511

**Comment by elukey** [comment]
@justiceotuya @ragesoss Hi! A fix has been deployed to support the Content-Type header in CORS policies, could you please re-test?

**Comment by justiceotuya** [comment]
@ragesoss fixed

File: config/routes.rb [hunk 1]
```diff
@@ -302,6 +302,7 @@
 
   # Revision Feedback
   get '/revision_feedback' => 'revision_feedback#index'
+  # post '/revision_feedback' => 'revision_feedback#index'
```
Comment (ragesoss) [inline config/routes.rb:305]: This is not needed.

**Comment by ragesoss** [comment]
It looks like the spec that tests Article Finder is still failing.

**Comment by ragesoss** [comment]
It looks like this is basically working, although in testing it locally I'm getting some intermittent errors that are not handled well. (On the LiftWing side, I saw one instance of a 504 error, which caused a red error notification but showed no useful user-facing information. There are also some errors that sometimes happen with the pageviews API, which are probably related to #5508. I'll open a new issue for these once this is merged.)

It looks like this PR combines a lot of code reformatting with the actual changes to move to LiftWing, which makes it a lot harder for me to review. Could you isolate the necessary changes without any formatting changes alongside it, so that I can think through the specific LiftWing-related code before merging? Thanks!

**Pull request closed without merging**